oauth2/handler.go

```diff
@@ -49,6 +65,20 @@ func (h *Handler) SetRoutes(r *httprouter.Router) {
 	r.GET(ConsentPath, h.DefaultConsentHandler)
 	r.POST(IntrospectPath, h.IntrospectHandler)
 	r.POST(RevocationPath, h.RevocationHandler)
+	r.GET(WellKnownPath, h.WellKnownHandler)
+}
+
+func (h *Handler) WellKnownHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	known := WellKnown{
+		Issuer:        h.Issuer,
+		AuthURL:       h.Issuer + AuthPath,
```

Comment:
I don't feel comfortable about this, what if the issuer is formated as `http://foo.bar/foo/` and the AuthPath as `/foo`, then we get `http://foo.bar/foo//foo` as a result. For resolving this, see [this SO question](http://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join)